go/parquet/pqarrow/column_readers.go

```diff
@@ -546,6 +551,14 @@ func transferBinary(rdr file.RecordReader, dt arrow.DataType) *arrow.Chunked {
 			defer chunks[idx].Data().Release()
 			defer chunks[idx].Release()
 		}
+	} else if dt.ID() == arrow.EXTENSION && len(chunks) > 0 && arrow.StorageTypeEqual(chunks[0].DataType(), dt) && !arrow.TypeEqual(chunks[0].DataType(), dt) {
```

Comment:
this is kinda what i was touching on above. Because you're passing the extension data type down here rather than the expected underlying storage, this becomes a bit more complicated.

I think it might be preferable to do something like:

```go
chunks := brdr.GetBuilderChunks()
storage := dt
if dt.ID() == arrow.EXTENSION {
    storage = dt.(arrow.ExtensionType).StorageType()
}

if storage == arrow.BinaryTypes.String || storage == arrow.BinaryTypes.LargeString {
    // if storage != dt -> do the loop with `NewExtensionArrayWithStorage` 
    // otherwise the look with `MakeFromData`
}
return arrow.NewChunked(dt, chunks)
```

Reply:
The resulting code seems harder to follow as both cases are now somewhat intertwined.